packages/agent/src/did-api.ts

```diff
@@ -0,0 +1,321 @@
+  DidDocument,
+  DidMetadata,
+  PortableDid,
+  DidMethodApi,
+  DidResolverCache,
+  DidDhtCreateOptions,
+  DidJwkCreateOptions,
+  DidResolutionResult,
+  DidResolutionOptions,
+  DidVerificationMethod,
+} from '@web5/dids';
+
+import { BearerDid, Did, DidResolver } from '@web5/dids';
+
+import type { AgentDataStore } from './store-data.js';
+import type { AgentKeyManager } from './types/key-manager.js';
+import type { ResponseStatus, Web5PlatformAgent } from './types/agent.js';
+
+import { InMemoryDidStore } from './store-did.js';
+import { DidResolverCacheMemory } from './prototyping/dids/resolver-cache-memory.js';
+
+export enum DidInterface {
+  Create  = 'Create',
+  // Deactivate = 'Deactivate',
+  Resolve = 'Resolve',
+  // Update  = 'Update'
+}
+
+export interface DidMessageParams {
+  [DidInterface.Create]: DidCreateParams;
+  // [DidInterface.Deactivate]: DidDeactivateParams;
+  [DidInterface.Resolve]: DidResolveParams;
+  // [DidInterface.Update]: DidUpdateParams;
+}
+
+export interface DidMessageResult {
+  [DidInterface.Create]: DidCreateResult;
+  // [DidInterface.Deactivate]: DidDeactivateResult;
+  [DidInterface.Resolve]: DidResolveResult;
+  // [DidInterface.Update]: DidUpdateResult;
+}
+
+export type DidCreateResult = {
+  uri: string;
+  document: DidDocument;
+  metadata: DidMetadata;
+}
+
+export type DidResolveResult = DidResolutionResult
+
+export type DidRequest<T extends DidInterface> = {
+  messageType: T;
+  messageParams: DidMessageParams[T];
+}
+
+export type DidResolveParams = {
+  didUri: string;
+  options?: DidResolutionOptions;
+}
+
+export type DidResponse<T extends DidInterface> = ResponseStatus & {
+  result?: DidMessageResult[T];
+};
+
+export interface DidCreateParams<
+  TKeyManager = AgentKeyManager,
+  TMethod extends keyof DidMethodCreateOptions<TKeyManager> = keyof DidMethodCreateOptions<TKeyManager>
+> {
+  method: TMethod;
+  options?: DidMethodCreateOptions<TKeyManager>[TMethod];
+  store?: boolean;
+  tenant?: string;
+}
+
+export interface DidMethodCreateOptions<TKeyManager> {
+  dht: DidDhtCreateOptions<TKeyManager>;
+  jwk: DidJwkCreateOptions<TKeyManager>;
+}
+
+export interface DidApiParams {
+  didMethods: DidMethodApi[];
+
+  agent?: Web5PlatformAgent;
+
+  /**
+   * An optional `DidResolverCache` instance used for caching resolved DID documents.
+   *
+   * Providing a cache implementation can significantly enhance resolution performance by avoiding
+   * redundant resolutions for previously resolved DIDs. If omitted, a no-operation cache is used,
+   * which effectively disables caching.
+   */
+  resolverCache?: DidResolverCache;
+
+  store?: AgentDataStore<PortableDid>;
+}
+
+export function isDidRequest<T extends DidInterface>(
+  didRequest: DidRequest<DidInterface>, messageType: T
+): didRequest is DidRequest<T> {
+  return didRequest.messageType === messageType;
+}
+
+export class AgentDidApi<TKeyManager extends AgentKeyManager = AgentKeyManager> extends DidResolver {
```

Comment:
A comment here would help a user understand why they might need (or not need) this class